integration_tests/test_competition_download.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from requests import HTTPError
+
+from kagglehub import competition_download
+
+from .utils import assert_files, create_test_cache
+
+HANDLE = "titanic"
+
+
+class TestCompetitionDownload(unittest.TestCase):
+    def test_competition_succeeds(self) -> None:
+        with create_test_cache():
+            expected_files = [
+                "gender_submission.csv",
+                "test.csv",
+                "train.csv",
+            ]
+
+            actual_path = competition_download(HANDLE)
+
+            assert_files(self, actual_path, expected_files)
+
+    def test_competition_login_needed_succeeds(self) -> None:
+        with create_test_cache():
+            expected_files = [
+                "sample_submission.csv",
+                "test_identity.csv",
+                "test_transaction.csv",
+                "train_identity.csv",
+                "train_transaction.csv",
+            ]
+
+            actual_path = competition_download("ieee-fraud-detection")
+
+            assert_files(self, actual_path, expected_files)
+
+    def test_competition_multiple_files(self) -> None:
+        with create_test_cache():
+            file_paths = [
+                "gender_submission.csv",
+                "test.csv",
+                "train.csv",
+            ]
+            for p in file_paths:
```

Comment:
What error is shown to the user if they haven't accepted the competition rules?

Reply:
authentication error: 403 

i've updated tests to better capture the error that supposed to be given.

Reply:
Ideally, we would show a better error message so the user can understand why they can't download the dataset and what they must do to get access.

Reply:
currently the error message is identical to that of dataset and model:
https://screenshot.googleplex.com/47Ci23jVYpb6uYK

"You don't have permission to access resource at URL: https://www.kaggle.com/competitions/m5-forecasting-accuracy
Please make sure you are authenticated if you are trying to access a private resource or a resource requiring consent."


would we want to add an additional: 
"please accept competition rule" something along those lines?

Reply:
Yes, and maybe with a link to where to go to accept the rules: https://www.kaggle.com/competitions/m5-forecasting-accuracy/rules

And of course, we should only show the message about accepting rules for competition datasets and not other types of resources.

Reply:
d.